src/arr/compiler/pyret.arr

```diff
@@ -145,31 +153,38 @@ fun main(args):
               display-progress: display-progress
             })
           failures = filter(CS.is-err, result.loadables)
-          when is-link(failures):
+          if is-link(failures):
             for each(f from failures) block:
               for lists.each(e from f.errors) block:
                 print-error(tostring(e))
                 print-error("\n")
               end
-              raise("There were compilation errors")
+              _ = print("There were compilation errors\n")
+              failure-code
             end
+          else:
+            success-code
           end
         else if r.has-key("run"):
-          CLI.run(r.get-value("run"), CS.default-compile-options.{
+          _ = CLI.run(r.get-value("run"), CS.default-compile-options.{
               standalone-file: standalone-file,
               compile-module: true,
               display-progress: display-progress,
               check-all: check-all
             })
+          success-code
         else:
-          print(C.usage-info(options).join-str("\n"))
-          raise("Unknown command line options")
+          _ = print(C.usage-info(options).join-str("\n"))
+          _ = print("Unknown command line options\n")
+          failure-code
```

Comment:
Cool! Here is what I am thinking.

- If using the main `pyret.arr` results in having a stack trace (Pyret or JS) in the output, it's either because the user supplied some kind of `--verbose-errors` option, or there is a legitimate bug that would need to get resolved, or maybe both.
- Otherwise, if there is some user error, we should print a friendly/helpful message.
- If `--verbose-errors` is _not_ supplied, then a non-zero exit code should only result either from such anticipated user errors, or due to failing `check` statements.

There may be cases I didn't consider, but I feel like the general spirit of the above list seems decent.